### PR TITLE
crio 1.12 and ssh-bastion in a pod

### PR DESCRIPTION
**Pull request by vrutkovs**
Supersedes #11167

Setup crio 1.12 and ssh bastion in a pod to reach scaled up RHEL nodes

```
Failing tests:

The bootstrap user should successfully login with password decoded from kubeadmin secret [Suite:openshift/conformance/parallel]
2x [sig-api-machinery] Namespaces [Serial] should ensure that all pods are removed when a namespace is deleted [Conformance] [Suite:openshift/conformance/serial/minimal] [Suite:k8s]
```

### CODE REVIEW

File: roles/container_runtime/tasks/package_crio.yml [hunk 1]
```diff
@@ -46,11 +46,19 @@
...
+    value: '"{{ openshift_crio_pause_image }}"'
+
+- name: Set pause_command to pod cmd
+  ini_file:
     dest: /etc/crio/crio.conf
-    src: crio.conf.j2
-    backup: yes
+    section: "crio.image"
+    option: pause_command
+    value: ' "/usr/bin/pod"'
```
Comment (DanyC97) [inline roles/container_runtime/tasks/package_crio.yml]: you didn't like the template ?

imo a template would give you a much better flexibility in the long run when things will get much more stable and you will realize you want to do some tuning.

**Comment by DanyC97** [comment]
Vadim, love this ssh bastion pod, is a killer man ! ;)

File: roles/container_runtime/tasks/package_crio.yml (hunk 1 above)
Comment (vrutkovs) [reply roles/container_runtime/tasks/package_crio.yml]: We don't directly control the crio versions, unlike docker, so its best to modify just the required parts

**Comment by mtnbikenc** [comment]
Looks like a few things changed, need to update:
`machinesets.machine.openshift.io` -> `machinesets`
`openshift-machine-api` -> `openshift-cluster-api`

```
$ oc get machinesets -o name -n openshift-cluster-api
machineset.cluster.k8s.io/rteague-worker-us-east-2a
machineset.cluster.k8s.io/rteague-worker-us-east-2b
machineset.cluster.k8s.io/rteague-worker-us-east-2c
```

**Comment by mtnbikenc** [comment]
Also:
`machines.machine.openshift.io` -> `machines`

File: test/aws/files/01_service.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
```
Comment (sdodson) [inline test/aws/files/01_service.yml]: If there's any chance we'll drop more stuff in here should we name them 01_ssh_service.yml, 02_ssh_serviceaccount.yml, etc?

Comment (vrutkovs) [reply test/aws/files/01_service.yml]: Right, `oc apply` applies files in alphabetical order, so first the namespace has to be created, then a service etc.

**Comment by mtnbikenc** [comment]
My two comments above about needing to change machines and machinesets was due to the fact I was using the 0.12 installer.  After building the installer from master I didn't have those issues.

File: roles/container_runtime/tasks/package_crio.yml (hunk 1 above)
Comment (DanyC97) [reply roles/container_runtime/tasks/package_crio.yml]: right, i was coming from the angle where later the default CRI-O settings won't satisfy the needs of BYO .. i guess we can cross that bridge when we come to it

**Comment by sdodson** [comment]
/lgtm

As a team we feel that this code is generally working as intended, however we're running into persistent e2e test failures that we'll need to address with other teams. Merging as is.

**Pull request merged**